### PR TITLE
Updates inlineCompletions proposal from inlineCompletionsNew proposal.

### PR DESCRIPTION
**Pull request by hediet**

### CODE REVIEW

File: src/vs/workbench/api/common/extHost.api.impl.ts [hunk 1]
```diff
@@ -526,6 +526,9 @@ export function createApiFactoryAndRegisterActors(accessor: ServicesAccessor): I
 			},
 			registerInlineCompletionItemProvider(selector: vscode.DocumentSelector, provider: vscode.InlineCompletionItemProvider): vscode.Disposable {
 				checkProposedApiEnabled(extension, 'inlineCompletions');
+				if (provider.handleDidShowCompletionItem && !isProposedApiEnabled(extension, 'inlineCompletionsAdditions')) {
+					throw new Error(`When the method "handleDidShowCompletionItem" is implemented on a provider, the usage of the proposed api 'inlineCompletionsAdditions' must be declared!`);
+				}
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHost.api.impl.ts:531]: Prefer to use the standard error that is thrown from `checkProposedApiEnabled`. That includes instructions how to enable a proposal. Alternatively, only warn about the `handleDidShowCompletionItem`-property and then ignore it

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts [hunk 2]
```diff
@@ -1033,17 +1032,27 @@ class InlineCompletionAdapterBase {
 }
 
 class InlineCompletionAdapter extends InlineCompletionAdapterBase {
-	private readonly _cache = new Cache<vscode.InlineCompletionItem>('InlineCompletionItem');
-	private readonly _disposables = new Map<number, DisposableStore>();
+	private readonly _references = new ReferenceMap<{
+		dispose(): void;
+		items: readonly vscode.InlineCompletionItem[];
+	}>();
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostLanguageFeatures.ts:1038]: Nit but I would preferred to see `Cache` being used. For the sake for code-reuse

File: src/vs/workbench/api/common/extHostTypes.ts [hunk 3]
```diff
@@ -1595,33 +1595,24 @@ export class CompletionList {
...
-	insertText?: string | SnippetString;
-
-	filterText?: string;
-
-	/**
-	 * @deprecated Use `insertText` instead. Will be removed eventually.
-	*/
-	text?: string;
-
+	insertText: string;
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostTypes.ts:1598]: `filterText` got lost

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts [hunk 4]
```diff
@@ -1084,41 +1099,40 @@ class InlineCompletionAdapter extends InlineCompletionAdapterBase {
...
 					insertText: typeof insertText === 'string' ? insertText : { snippet: insertText.value },
+					filterText: item.filterText,
 					range: item.range ? typeConvert.Range.from(item.range) : undefined,
 					command,
 					idx: idx,
-					completeBracketPairs: item.completeBracketPairs
+					completeBracketPairs: this.isAdditionProposedApiEnabled ? item.completeBracketPairs : false
 				});
 			}),
+			commands: commands.map(c => {
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostLanguageFeatures.ts:1116]: Make sure to use `this.isAdditionProposedApiEnabled` here as well

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts [hunk 5]
```diff
@@ -1084,41 +1099,40 @@ class InlineCompletionAdapter extends InlineCompletionAdapterBase {
...
-		const completionItem = this._cache.get(pid, idx);
+		const completionItem = this._references.get(pid)?.items[idx];
 		if (completionItem) {
-			InlineCompletionController.get(this._provider).fireOnDidShowCompletionItem({
-				completionItem
-			});
+			if (this._provider.handleDidShowCompletionItem && this.isAdditionProposedApiEnabled) {
+				this._provider.handleDidShowCompletionItem(completionItem);
+			}
 		}
```
Comment (jrieken) [inline src/vs/workbench/api/common/extHostLanguageFeatures.ts:1136]: When registering an inline completion item provider you can pass along to the renderer if the provider supports `handleDidShow...`. With that you can skip the IPC call for most providers. See `this._proxy.$registerCompletionsProvider` and others for sample cases

**Review by jrieken — APPROVED** [review]

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts (hunk 5 above)
Comment (hediet) [reply src/vs/workbench/api/common/extHostLanguageFeatures.ts:1136]: Should I just check for `this.isAdditionProposedApiEnabled` or if `handleDidShowCompletionItem` is set?
The field `handleDidShowCompletionItem` could be initialized later.

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts (hunk 2 above)
Comment (jrieken) [reply src/vs/workbench/api/common/extHostLanguageFeatures.ts:1038]: Oh, didn't notice that. Maybe a case for wider adoption. I have used an ugly "siblings" pattern before, e.g [here](https://github.com/microsoft/vscode/blob/93fd393a0e200f3cc8038cef0e61b14a75fb1c21/src/vs/workbench/api/common/extHostLanguageFeatures.ts#L875-L876)

File: src/vs/workbench/api/common/extHostLanguageFeatures.ts (hunk 5 above)
Comment (jrieken) [reply src/vs/workbench/api/common/extHostLanguageFeatures.ts:1136]: > The field handleDidShowCompletionItem could be initialized later.

We ignore that and for consistence reasons should continue to do so. If it's not there at `register`-time than it is never there

**Review by jrieken — APPROVED** [review]

**Review by jrieken — APPROVED** [review]

**Review by jrieken — APPROVED** [review]

**Review by jrieken — APPROVED** [review]

**Pull request merged**